hack/generator/pkg/config/type_filter.go

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package config
+
+// TypeFilterAction defines the possible actions that should happen for types matching the filter
+type TypeFilterAction string
+
+const (
+	// TypeFilterIncludeType indicates that any type matched by the filter should be included in the type graph
+	TypeFilterIncludeType TypeFilterAction = "include"
+	// TypeFilterPruneType indicates that any type matched by the filter, and any types only referenced by that type
+	// should not be included in the type graph
+	TypeFilterPruneType TypeFilterAction = "prune"
```

Comment:
If we keep the longer naming, should use `TypeFilterAction` as the prefix: `TypeFilterActionPrune` etc